### PR TITLE
style: Apply PEP 585/604 for type annotations

### PR DESCRIPTION
**Pull request by tachikoma-li**
Changes in this PR:
- Apply the PEP standard as discussed.
- Add missing type hint.
- Minor tweaks to some docs.

### CODE REVIEW

File: qctrlopencontrols/utils.py [hunk 1]
```diff
@@ -81,30 +87,38 @@ def create_repr_from_attributes(class_instance=None, attributes=None):
...
 
         def log(x):
-            check_arguments(x > 0,
-                            "x must be positive.",
-                            {"x": x})
-            return numpy.log(x)
+            check_arguments(
+                x > 0,
+                "x must be positive.",
+                {"x": x}
```
Comment (abenseny) [inline qctrlopencontrols/utils.py:106]:
```suggestion
                x > 0, "x must be positive.", {"x": x}
```
or
```suggestion
                x > 0,
                "x must be positive.",
                {"x": x},
```
?

**Review by perjlieb — DISMISSED** [review]
🗄️

**Review by abenseny — APPROVED** [review]
Thanks Li!
🪥

File: qctrlopencontrols/utils.py (hunk 1 above)
Comment (tachikoma-li) [reply qctrlopencontrols/utils.py]: Haha, yeah, black.

**Review by abenseny — APPROVED** [review]
✨

**Pull request merged**